neurots/generate/tree.py

```diff
@@ -248,8 +262,21 @@ def next_point(self):
         for section_grower in ordered_sections:
             # the current section_grower is generated
             # In here the stop criterion can be modified accordingly
+
+            if "cut_major_length" in self.params:  # pragma: no cover
+                # this makes an early termination of major branch
+                # it needs to revert the value after getting the state to preserve
+                # the original topology
+                if section_grower.process == "major":
+                    _term = section_grower.stop_criteria["TMD"].term
+                    section_grower.stop_criteria["TMD"].term = self.params["cut_major_length"]
+
             state = self.growth_algo.extend(section_grower)
 
+            if "cut_major_length" in self.params:  # pragma: no cover
+                if section_grower.process == "major":
+                    section_grower.stop_criteria["TMD"].term = _term
```

Comment:
This is already in #78

Reply:
ah ok, so #78 should be merge before this one and then you'll rebase?

Reply:
yes, so i can use both together